sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -97,6 +97,26 @@ public static Throwable mapThrowableToTableServiceException(Throwable throwable)
         }
     }
 
+    /**
+     * Map a {@link Throwable} to {@link IllegalArgumentException} if the endpoint
+     * the table name, else it returns the original throwable
+     * @param throwable A throwable exception
+     * @param tableName The name of the Table intended to be operated on
+     * @param endpoint The endpoint of the {@link com.azure.data.tables.TableServiceClient},
+     *                 {@link com.azure.data.tables.TableServiceAsyncClient}
+     * @return A Throwable that is either an instance of {@link IllegalArgumentException} or the original throwable.
+     */
+    public static Throwable mapThrowableToIllegalArgumentExceptionWhenTableNamePresent(Throwable throwable,
+                                                                                       String tableName,
+                                                                                       String endpoint) {
+        if (endpoint.contains(tableName)) {
```

Comment:
This would probably be a very odd case, but if the host happens to share the same name as the table (or even partially) this check would act on a "false positive". For example: https://mytablesaccount.core.windows.com/mytable. We should probably check against the URL's path instead of the whole URL.

Reply:
Ok, I'll change it to look at the suffix.